src/components/ExerciseHeader.tsx

```diff
@@ -57,21 +58,29 @@ const ExerciseHeader = ({
   confirmClose = true,
   labelOverride,
   isCloseButton = true,
+  exerciseKey,
 }: ExerciseHeaderProps): JSX.Element => {
   const [isModalVisible, setIsModalVisible] = useState(false)
   const [isFeedbackModalVisible, setIsFeedbackModalVisible] = useState(false)
   const theme = useTheme()
   const showProgress = numberOfWords !== undefined && numberOfWords > 0 && currentWord !== undefined
   const progressText = showProgress ? `${currentWord + 1} / ${numberOfWords}` : ''
-
+  const isWordList = exerciseKey === 0 || exerciseKey === undefined
   useEffect(() => {
-    const renderHeaderLeft = () => (
-      <NavigationHeaderLeft
-        title={labelOverride ?? getLabels().general.header.cancelExercise}
-        onPress={confirmClose ? () => setIsModalVisible(true) : () => navigation.dispatch(closeExerciseAction)}
-        isCloseButton={isCloseButton}
-      />
-    )
+    const renderHeaderLeft = () =>
+      isWordList && numberOfWords === (currentWord ?? 0) + 1 ? (
```

Comment:
If you add more variables like suggested below you don't need the if block here and define the `NavigationHeaderLeft` twice. Imo that makes it a lot more readable/less code.
``` ts
  const isLastWordInList = isWordList && numberOfWords === (currentWord ?? 0) + 1
  const title = labelOverride ?? isLastWordInList ? getLabels().general.header.backToWordlist : getLabels().results.action.cancelExercise
```

Reply:
ok, that is way better. Thank you, I will change it

Reply:
@steffenkleinle what if i need to include onPress as well. I think it is not possible then to manage it without defining it twice.